src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1351,6 +1359,12 @@ void primitive_inst::do_runtime_in_place_concat() {
         ++i;
     }
     concat_inst->_impl_params->output_layouts[0] = concat_layout; // TODO : Once this primitive_inst::can_be_optimized, consolidate it to impl_params->optimized
+
+    if (has_subgraph_dependency(concat_inst->dependencies())) {
```

Comment:
Please move this check to the top of this function, because code above could change predcessor's padding before existing the function by your condition